frontend/src/workspaces/features/FilesEditor/FilesEditor.tsx

```diff
@@ -0,0 +1,208 @@
+      >
+        {isExpanded ? (
+          <ChevronDownIcon className="w-4 h-4 mr-2 text-gray-400" />
+        ) : (
+          <ChevronRightIcon className="w-4 h-4 mr-2 text-gray-400" />
+        )}
+        <FolderIcon className="w-4 h-4 mr-2 text-gray-400" />
+        <span>{node.name}</span>
+      </div>
+      {isExpanded && node.children && (
+        <div>
+          {node.children.map((child) => (
+            <FileTreeNode
+              key={child.path}
+              node={child}
+              level={level + 1}
+              selectedFile={selectedFile}
+              setSelectedFile={setSelectedFile}
+            />
+          ))}
+        </div>
+      )}
+    </div>
+  );
+};
+
+export type FileNode = FilesEditor_FileFragment & {
+  children: FileNode[];
+};
+
+interface FilesEditorProps {
+  name: string;
+  files: FilesEditor_FileFragment[];
+}
+export const FilesEditor = ({ name, files: flatFiles }: FilesEditorProps) => {
+  const { t } = useTranslation();
+  const files = useMemo(() => {
+    return buildTreeFromFlatData(flatFiles);
+  }, [flatFiles]);
+  const rootFiles = useMemo(() => {
+    return files.filter((file) => !file.parentId);
+  }, [files]);
+
+  const [selectedFile, setSelectedFile] = useState<FileNode | null>(
+    files.filter((file) => file.autoSelect)[0] || null,
+  );
+
+  const numberOfFiles = files.filter((file) => file.type === "file").length;
+
+  return (
+    <div className="flex border border-gray-200 rounded-lg overflow-hidden min-h-[400px] max-h-[75vh]">
+      <div className="w-80 bg-gray-50 border-r border-gray-200 flex flex-col">
+        <div className="p-3 border-b border-gray-200 bg-white flex-shrink-0">
+          <h3 className="text-sm font-medium text-gray-900">
+            {t("Files")} - {name}
+          </h3>
+          <div className="text-xs text-gray-500 mt-1">
+            {numberOfFiles} {t("files")}
+          </div>
+        </div>
+        <div className="py-2 overflow-y-auto flex-1">
+          {rootFiles.map((node) => (
+            <FileTreeNode
+              key={node.path}
+              node={node}
+              selectedFile={selectedFile}
+              setSelectedFile={setSelectedFile}
+            />
+          ))}
+        </div>
+      </div>
+
+      <div className="flex-1 flex flex-col min-w-0">
+        {selectedFile ? (
+          <>
+            <div className="p-3 border-b border-gray-200 bg-white flex-shrink-0">
+              <div className="text-sm font-medium text-gray-900">
+                {selectedFile.name}
+              </div>
+              <div className="text-xs text-gray-500 mt-1">
+                {selectedFile.language}
+                {" • "}
+                {selectedFile.lineCount}
+                {` ${t("lines")}`}
+              </div>
+            </div>
+            <div className="flex-1 overflow-hidden">
+              <CodeMirror
+                value={selectedFile.content!}
+                readOnly={true}
```

Comment:
Really exiting :) Seems to me like making editing possible will be a relative low effort by using `CodeMirror`, great choice :+1: